### ISSUE TITLE
get wrong result when using sparse_categorical_crossentropy in custom loss function

### ISSUE THREAD

**Issue by TuXiaokang** [fchollet/keras#6330]
hi, when i write code like this
```python

def custom_loss(y_true, y_pred):
    return K.sparse_categorical_crossentropy(y_pred, y_true)

model.compile(loss=custom_loss,
              optimizer=keras.optimizers.Adadelta(),
              metrics=['accuracy'])
```
i have got difference result with

```python

model.compile(loss='sparse_categorical_crossentropy',
              optimizer=keras.optimizers.Adadelta(),
              metrics=['accuracy'])
```
I'm a little confused, How could this be??

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
